src/app/zap-templates/partials/cluster-objects-struct.zapt

```diff
@@ -78,15 +78,22 @@ CHIP_ERROR Type::EncodeForRead(TLV::TLVWriter &writer, TLV::Tag tag, FabricIndex
 
 CHIP_ERROR Type::DoEncode(TLV::TLVWriter &writer, TLV::Tag tag, const Optional<FabricIndex> & accessingFabricIndex) const
 {
+    {{#if struct_has_sensitive_fields}}
+    bool includeSensitive = !accessingFabricIndex.HasValue() || (accessingFabricIndex.HasValue() && accessingFabricIndex.Value() == {{ asLowerCamelCase struct_fabric_idx_field }});
```

Comment:
```suggestion
    bool includeSensitive = !accessingFabricIndex.HasValue() || (accessingFabricIndex.Value() == {{ asLowerCamelCase struct_fabric_idx_field }});
```
because HasValue will definitely be true on the RHS of `||` (otherwise we would never have gotten there.